### PR TITLE
Reader Full Post: show a friendly message for 403 or 404 errors

### PR DESCRIPTION
**Pull request by bluefuton**
Fixes https://github.com/Automattic/wp-calypso/issues/8989.

This PR adds a helpful message when a full post is missing or unavailable. It also outputs the error if in a development environment:

screen shot 2016-12-05 at 17 04 07

I've used one general message ("Post unavailable") to address all errors for the moment.

### To test 

Try visiting a post that doesn't exist, and ensure you see the 'Post unavailable' page.

Try visiting a post that you don't have permission to view (e.g. a private P2 with an unprivileged account), and ensure you see the 'Post unavailable' page.

Make sure the Back button works, both on a regular full post and on the 'Post unavailable' page.

### CODE REVIEW

File: client/blocks/reader-full-post/index.jsx [hunk 1]
```diff
@@ -211,6 +211,10 @@ export class FullPostView extends React.Component {
 	}
 
 	parseEmoji() {
+		if ( ! this.refs.article ) {
```
Comment (samouri) [inline client/blocks/reader-full-post/index.jsx]: why was this added?

File: client/lib/feed-post-store/index.js [hunk 2]
```diff
@@ -83,7 +83,7 @@ FeedPostStore.dispatchToken = Dispatcher.register( function( payload ) {
 		case FeedPostActionType.RECEIVE_FEED_POST:
 			if ( action.error ) {
 				const error = {
-					status_code: -1,
+					status_code: action.error.statusCode ? action.error.statusCode : -1,
```
Comment (samouri) [inline client/lib/feed-post-store/index.js]: when would `action.error` be truthy but `action.error.statusCode` be falsy?

**Comment by samouri** [comment]
Tested:
- [x] visited fake post ids for real blogs and ensured I saw the new 404 post unavailable screen in with refresh flag on/off.
- [x] visited restricted blog with on account without access

Assuming we don't change anything based on the question I asked above I say  :shipit:

File: client/blocks/reader-full-post/index.jsx (hunk 1 above)
Comment (bluefuton) [inline client/blocks/reader-full-post/index.jsx]: There's an attempt to parse emoji on `ReaderFullPostUnavailable` otherwise, which fails because the `ref` is missing:

screen shot 2016-12-06 at 11 27 37

File: client/lib/feed-post-store/index.js (hunk 2 above)
Comment (bluefuton) [inline client/lib/feed-post-store/index.js]: Just being cautious here. It looks from [receiveError](https://github.com/Automattic/wp-calypso/blob/5f44fdb50e31aa54161fa8ad96cabd6363c7fb98/client/lib/feed-post-store/index.js#L235) that not finding a status code is a possibility.
